src/bit-systems/billboard.ts

```diff
@@ -0,0 +1,78 @@
+import { defineQuery } from "bitecs";
+import { Box3, Frustum, Matrix4, Vector3, Object3D, Camera, Mesh } from "three";
+import { HubsWorld } from "../app";
+import { Billboard } from "../bit-components";
+
+const billboardQuery = defineQuery([Billboard]);
+
+const isMobileVR = AFRAME.utils.device.isMobileVR() || true;
+
+const targetPos = new Vector3();
+const worldPos = new Vector3();
+const frustum = new Frustum();
+const frustumMatrix = new Matrix4();
+const box = new Box3();
+const boxTemp = new Box3();
+
+const expandBox = (child: Mesh) => {
+  if (child.geometry) {
+    child.updateMatrices();
+    child.geometry.computeBoundingBox();
+    boxTemp.copy(child.geometry.boundingBox!).applyMatrix4(child.matrixWorld);
+    box.expandByPoint(boxTemp.min);
+    box.expandByPoint(boxTemp.max);
+  }
+};
+
+const isInViewOfCamera = (object3D: Object3D, camera: Camera) => {
+  frustumMatrix.multiplyMatrices(camera.projectionMatrix, camera.matrixWorldInverse);
+  frustum.setFromProjectionMatrix(frustumMatrix);
+  box.makeEmpty();
+  object3D.traverse(expandBox);
+
+  // NOTE: not using box.setFromObject here because text nodes do not have Z values in their geometry buffer,
+  // and that routine ultimately assumes they do.
+  return frustum.intersectsBox(box);
+};
+
+const updateIsInView = (world: HubsWorld, billboard: number, camera: Camera): boolean => {
```

Comment:
nit: Might call this something like `shouldUpdateBillboard` since it's not storing `inView` state in a map

Reply:
Oh yep, makes sense.